### PR TITLE
Updated package files to recognize python 3 dependencies.

### PR DESCRIPTION
**Pull request by krp97**
Introduced the following changes:
- Updated the package/pkg-python.mk, in order to recognize **HOST_PKG_NEEDS_HOST_PYTHON** switch.
- Updated the chain of dependencies of wpeframework-tools.

### CODE REVIEW

**Comment by krp97** [comment]
The need for these changes can be found in a [porting generator scripts to python 3 PR.](https://github.com/WebPlatformForEmbedded/Thunder/pull/333)

**Comment by bramoosterhuis** [comment]
Looks like you got inspired by the upstream buildroot :+1:

File: package/python3-setuptools/python3-setuptools.mk [hunk 1]
```diff
@@ -0,0 +1,9 @@
+PYTHON3_SETUPTOOLS_VERSION = 18.7.1
+PYTHON3_SETUPTOOLS_SOURCE = setuptools-$(PYTHON_SETUPTOOLS_VERSION).tar.gz
+PYTHON3_SETUPTOOLS_SITE = http://pypi.python.org/packages/source/s/setuptools
+PYTHON3_SETUPTOOLS_LICENSE = Python Software Foundation or Zope Public License
+PYTHON3_SETUPTOOLS_LICENSE_FILES = PKG-INFO
+PYTHON3_SETUPTOOLS_SETUP_TYPE = setuptools
+HOST_PYTHON3_SETUPTOOLS_NEEDS_HOST_PYTHON = python3
+
+$(eval $(host-python-package))
```
Comment (artur-gebicz-ml) [inline package/python3-setuptools/python3-setuptools.mk]: Please add newline

File: package/python3-jsonref/Config.in.host [hunk 2]
```diff
@@ -0,0 +1,7 @@
+config BR2_PACKAGE_HOST_PYTHON3_JSONREF
```
Comment (artur-gebicz-ml) [inline package/python3-jsonref/Config.in.host]: Please remove that file

File: package/python-jsonref/python-jsonref.mk [hunk 3]
```diff
@@ -10,6 +10,6 @@ PYTHON_JSONREF_SITE = https://pypi.python.org/packages/b3/cf/93d4f34d76863d4fb99
 PYTHON_JSONREF_LICENSE = MIT
 PYTHON_JSONREF_LICENSE_FILES = COPYING json/LICENSE
 PYTHON_JSONREF_SETUP_TYPE = setuptools
-PYTHON_JSONREF_DEPENDENCIES = host-python-setuptools
+PYTHON_JSONREF_DEPENDENCIES = host-python host-python-setuptools
```
Comment (artur-gebicz-ml) [inline package/python-jsonref/python-jsonref.mk]: Those dependencies shouldn't be here. Please remove them

File: package/pkg-python.mk [hunk 4]
```diff
@@ -20,14 +20,18 @@
 #
 ################################################################################
 
+define PKG_PYTHON_SYSCONFIGDATA_NAME
+$(basename $(notdir $(wildcard $(STAGING_DIR)/usr/lib/python$(PYTHON3_VERSION_MAJOR)/_sysconfigdata__linux_*.py)))
+endef
+
 # Target distutils-based packages
 PKG_PYTHON_DISTUTILS_ENV = \
```
Comment (artur-gebicz-ml) [inline package/pkg-python.mk]: Did you checked if any of packages, that are using distutils as setup method, are able to build? If not please do

**Comment by krp97** [comment]
Requested changes can be accounted for in this commit: 891743691f59eabec1517c7e15e8299b14b39707

Regarding distutils, a package: python-markdown2(with distutlis as a setup method) has been successfully built.

**Pull request merged**